internal/stanza/converter.go

```diff
@@ -582,3 +571,60 @@ var sevTextMap = map[entry.Severity]string{
 	entry.Fatal3:  "Fatal3",
 	entry.Fatal4:  "Fatal4",
 }
+
+// pairSep is chosen to be an invalid byte for a utf-8 sequence
+// making it more unlikely to be hit
+var pairSep = []byte{0xfe}
+
+func getResourceID(resource map[string]string) uint64 {
+	var fnvHash = fnv.New64a()
+	var fnvHashOut = make([]byte, 0, 16)
+	var keySlice = make([]string, 0, len(resource))
+	var escapedSlice = make([]byte, 0, 64)
+
+	for k := range resource {
+		keySlice = append(keySlice, k)
+	}
+
+	// In order for this to be deterministic, we need to sort the map. Using range, like above,
+	// has no guarantee about order.
+	sort.Strings(keySlice)
+	for _, k := range keySlice {
+		escapedSlice = appendEscapedPairSeparator(escapedSlice[:0], k)
+		fnvHash.Write(escapedSlice)
+		fnvHash.Write(pairSep)
+
+		escapedSlice = appendEscapedPairSeparator(escapedSlice[:0], resource[k])
+		fnvHash.Write(escapedSlice)
+		fnvHash.Write(pairSep)
+	}
+
+	fnvHashOut = fnvHash.Sum(fnvHashOut)
+	return binary.BigEndian.Uint64(fnvHashOut)
+}
+
+// appendEscapedPairSeparator escapes (prefixes) "pair_sep" and byte '0xff' with byte '0xff', and appends it to the
+// incoming buffer. It returns the appended buffer.
+func appendEscapedPairSeparator(buf []byte, s string) []byte {
+	const escapeByte byte = '\xff'
+
+	if len(s) > cap(buf) {
+		newBuf := make([]byte, len(s))
+		copy(newBuf, buf)
+		buf = newBuf
+	}
+
+	sBytes := []byte(s)
+	for _, b := range sBytes {
+		switch b {
+		case escapeByte:
+			fallthrough
+		case pairSep[0]:
```

Comment:
This might be a bit more readable, since there's no difference in processing:
```suggestion
		case escapeByte, pairSep[0]:
```

Reply:
I agree, way more readable.